### PR TITLE
fix: should override children if null is provided as an argument

### PR DESCRIPTION
**Pull request by clyfish**
https://github.com/facebook/react/blob/v17.0.2/packages/react/src/__tests__/ReactElement-test.js#L264-L273

React overrides children if null is provided as an argument, but preact doesn't.

### CODE REVIEW

**Comment by clyfish** [comment]
BTW, the new `jsx-runtime` also overrides children if null is provided as an argument.

**Comment by clyfish** [comment]
@developit  @marvinhagemeister  Please review this PR.

We are trying to switch from React to Preact, but this is a blocking issue.

React renders `<div children="A">{null}</div>` to `<div></div>`, but preact without jsx-runtime renders to `<div>A</div>`.

File: src/clone-element.js [hunk 1]
```diff
@@ -25,7 +25,7 @@ export function cloneElement(vnode, props, children) {
 			children.push(arguments[i]);
 		}
 	}
-	if (children != null) {
+	if (arguments.length > 2) {
```
Comment (marvinhagemeister) [inline src/clone-element.js:28]: Let's add a test case for the `cloneElement` case and than it's good to go 👍

**Review by marvinhagemeister — COMMENTED** [review]
Sorry for the late response. The PR is awesome! Left one minor comment and then it's ready to be merged 👍

**Review by JoviDeCroock — APPROVED** [review]

**Comment by marvinhagemeister** [comment]
Awesome, thank you so much for making a PR 🙌

**Comment by developit** [comment]
I think this causes `createElement('div', { children: 'foo' }, undefined)` to produce `{ children: undefined }`.

**Pull request merged**